htdocs/compta/facture/class/facture.class.php

```diff
@@ -2638,6 +2638,13 @@ public function insert_discount($idremise)
 			$facligne->multicurrency_total_tva = -$remise->multicurrency_amount_tva;
 			$facligne->multicurrency_total_ttc = -$remise->multicurrency_amount_ttc;
 
+			//Bugfix#32206 - Incorrect Discounts or Credit is applied in a multicurrency Invoice 
+			if ($this->multicurrency_tx > 0) {
```

Comment:
When multicurrency_tx is set, all other fields multicurrency should already contains a value include the rate. So bug shoul be at a higher place.